compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenOps.td

```diff
@@ -102,6 +103,59 @@ def IREECodegen_ExtractStridedMetadataOp : Op<IREECodegen_Dialect, "extract_stri
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// SwizzleHintOp
+//===----------------------------------------------------------------------===//
+
+def IREECodegen_SwizzleHintOp : Op<IREECodegen_Dialect, "swizzle_hint", [
+    SameOperandsAndResultType, Pure]> {
+  let summary = "Hint to swizzle accesses according to an access pattern";
+  let description = [{
+    Optimization hint to swizzle all accesses to the memref this takes a view
+    of. This only affects reads/writes immediately consuming this operation and
+    is best effort. If the desired swizzling is not apparently possible, this
+    op will no-op. As a result, it should not be relied on for correctness.
+
+    Note that this only rewrites direct users. If there are any aliased loads
+    or stores of the data from/to the |src| memref of a hintOp, those accesses
+    will not be swizzled. This allows reusing an allocation with different
+    swizzled access patterns as long as there is no data dependency between
+    memory with different layouts. For example:
+
+    ```
+    %0 = alloc()
+    %1 = iree_codegen.swizzle_hint %0, #layout_0
+    %2 = iree_codegen.swizzle_hint %0, #layout_1
+    {
+       vector.store %1
+       vector.load %1
+         ^
+         |
+        unrelated
+         |
+         v
+       vector.store %2
+       vector.load %2
+    }
+    ```
+
+    If there is a dependency between the accesses of %1 and %2, this is
```

Comment:
Clarify what kind of dependency you have in mind? I assume you mean memory dependencies -- in that case, should we say that this only works when %1 and %2 do not alias?